src/App.jsx

```diff
@@ -1,12 +1,141 @@
 import './App.css';
+import axios from 'axios';
+import { useState } from 'react';
+import worldMap from './assets/world_map.png';
+import getRegionColor from './helpers/helper';
+
 
 function App() {
+    const [countries, setCountries] = useState([]);
+    const [error, setError] = useState('');
+    const [loading, setLoading] = useState(false);
+    const [searchTerm, setSearchTerm] = useState('');
+    const [searchedCountry, setSearchedCountry] = useState(null);
+
+    async function fetchCountries() {
+        try {
+            setLoading(true);
+            const response = await axios.get('https://restcountries.com/v3.1/all');
+            const countriesData = response.data.map((country) => ({
+                name: country.name.common,
+                population: country.population || 'Unknown',
+            }));
+            const sortedCountries = countriesData.sort((a, b) =>
+                a.population === 'Unknown' || b.population === 'Unknown'
+                    ? 0
+                    : a.population - b.population
+            );
+            setCountries(sortedCountries);
+        } catch (e) {
+            console.log('De statuscode van de fout is:', e.response?.status);
+            setError('Het ophalen van de data is mislukt. Probeer het opnieuw');
+        } finally {
+            setLoading(false);
+        }
+    }
+
+    async function searchCountry() {
+        try {
+            setLoading(true);
+            const response = await axios.get(`https://restcountries.com/v3.1/name/{name}`);
+            setSearchedCountry(response.data[0]);
+            setError('');
+            setSearchTerm('');
+        } catch (e) {
+            setError(`${searchTerm} does not exist. Please try again.`);
+        } finally {
+            setLoading(false);
+        }
+    }
+
+    const handleSearch = async () => {
+        try {
+            await searchCountry();
+        } catch (error) {
+            console.error('Error during search:', error);
+        }
+    };
+
+    const handleInputChange = (event) => {
+        setSearchTerm(event.target.value);
+    };
+
+    const convertToMillions = (population) => {
+        return population ? `${(population / 1e6).toFixed(2)} million` : 'Unknown';
+    };
 
     return (
         <>
-            Maak je applicatie hier!
+            <header>
+                <img src={worldMap} alt="WorldMap" />
+                <h1>World Regions</h1>
+            </header>
+            <main>
+                <section>
+                    {!loading && (
+                        <button type="button" onClick={fetchCountries}>
+                            Haal landen op!
+                        </button>
+                    )}
+                    {loading && <p>Loading...</p>}
+                    {error && <p className="error">{error}</p>}
+                    {!loading && countries.length > 0 && (
+                        <ul>
+                            {countries.map((country) => (
```

Comment:
Je kunt hier nog spreading toepassen, zodat je niet overal `country.` vóór hoeft te zetten
```suggestion
                            {countries.map(({ name, region, population }) => (
```